### PR TITLE
Bbox in geo collection

### PR DESCRIPTION
**Pull request by bariela**
## Description

Adding bounding box in features in geojson collections as geoindexed field

## Motivation and Context

Features have often quite complex geometries, which are too complex for geoindexing functionality on mongodb,
bounding boxes have only four points in simple rectangle as shape

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have checked to ensure that there aren't other open [Pull Requests](../../../pulls) for the same update/change.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: initial-data-load/src/mapLayers/layersGeoDataUpload.js [hunk 1]
```diff
@@ -15,7 +17,9 @@ const { saveGeoJsonFromUrlSourceToStorage, saveGeoJsonFromFileToStorage } = requ
 
 const storeGeoDataToDb = async (fromFile, data, filePath) => {
   let geojsonData;
+  logger.info(`Clearing database collection ${data.collectionName}`);
   await deleteAllFromCollection(data.collectionName);
+  logger.info(`Loading data from file ${filePath}`);
```
Comment (srezacova) [inline initial-data-load/src/mapLayers/layersGeoDataUpload.js:22]: filePath is null when fetching data.

File: initial-data-load/src/mapLayers/layersGeoDataUpload.js [hunk 2]
```diff
@@ -26,8 +30,41 @@ const storeGeoDataToDb = async (fromFile, data, filePath) => {
...
   }
-  await createGeoDataIndex(data.collectionName);
-  await storeGeoFeaturesData(geojsonData.features, data.collectionName);
+  logger.info(`Creating index on collection...`);
+  await createGeoDataIndex(data.collectionName, 'bbox');
+  logger.info(`Generating bounding boxes for features...`);
+
+  const { features } = geojsonData;
+  if (!features || !features.length) {
+    logger.info(`No features in file ${filePath}`);
```
Comment (srezacova) [inline initial-data-load/src/mapLayers/layersGeoDataUpload.js:39]: filePath is null when fetching data.

File: initial-data-load/src/mapLayers/db.js [hunk 3]
```diff
@@ -10,8 +10,10 @@ const createIndex = async () => {
...
 
-const createGeoDataIndex = async (collectionName) => {
-  await mongoose.connection.db.collection(collectionName).createIndex({ geometry: '2dsphere' });
+const createGeoDataIndex = async (collectionName, geoKey = 'geometry') => {
+  const indexKey = {};
+  indexKey[geoKey] = '2dsphere';
+  await mongoose.connection.db.collection(collectionName).createIndex(indexKey);
 };
 
 const deleteAllFromCollection = async (collectionName) => {
```
Comment (srezacova) [inline initial-data-load/src/mapLayers/db.js:19]: There is warning in log:
initial-data-load-service_1  | (node:29) DeprecationWarning: collection.remove is deprecated. Use deleteOne, deleteMany, or bulkWrite instead.

maybe you can fix it in this task or leave it for fixing later.

File: initial-data-load/.gitignore [hunk 4]
```diff
@@ -4,3 +4,8 @@ node_modules
 .vscode
 spinup.sh
 .DS_Store
+
+/data
+!/data/Sample
```
[the remark this answers is not in the document]
Comment (bariela) [reply initial-data-load/.gitignore:9]: the test folders, we need included for tests     
and the Sample is, well, sample data,    
but anything else is not necessary, so better ignore it and not accidentally push public

File: initial-data-load/src/mapLayers/layersGeoDataUpload.js [hunk 5]
```diff
@@ -26,8 +30,41 @@ const storeGeoDataToDb = async (fromFile, data, filePath) => {
...
+    return;
+  }
+  const featuresWithBbox = geojsonData.features.map((feature) => {
```
[the remark this answers is not in the document]
Comment (bariela) [reply initial-data-load/src/mapLayers/layersGeoDataUpload.js:42]: thanks for catching it, will fix in next commit

File: initial-data-load/src/mapLayers/layersGeoDataUpload.js [hunk 6]
```diff
@@ -26,8 +30,41 @@ const storeGeoDataToDb = async (fromFile, data, filePath) => {
     }
     geojsonData = result.data;
   }
-  await createGeoDataIndex(data.collectionName);
-  await storeGeoFeaturesData(geojsonData.features, data.collectionName);
+  logger.info(`Creating index on collection...`);
+  await createGeoDataIndex(data.collectionName, 'bbox');
+  logger.info(`Generating bounding boxes for features...`);
+
+  const { features } = geojsonData;
```
[the remark this answers is not in the document]
Comment (bariela) [reply initial-data-load/src/mapLayers/layersGeoDataUpload.js:37]: if 'geojsonData' does not exist, the function is expected to fail at the data fetching parts above, so the issue with destructuring  is not there
and if features will be undefined, it's fine, as we check the case in the code next

File: initial-data-load/src/mapLayers/db.js (hunk 3 above)
Comment (bariela) [reply initial-data-load/src/mapLayers/db.js:19]: I would like to investigate the difference between 'remove' and the replacement before using one, especially in context of compatibility with CosmoDB. Definitely out of scope here

File: initial-data-load/src/mapLayers/layersGeoDataUpload.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (srezacova) [reply initial-data-load/src/mapLayers/layersGeoDataUpload.js:37]: yeah you are right, when features are undefined, its ok.

**Pull request merged**